gocwebtemplate-core/gocwebtemplate-core-base/src/main/java/goc/webtemplate/component/jsonentities/adapters/FooterLinkAdapter.java

```diff
@@ -0,0 +1,40 @@
+package goc.webtemplate.component.jsonentities.adapters;
+
+import java.io.IOException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import goc.webtemplate.FooterLinkContext;
+
+/**
+ * GSON type adapter for serializing the "privacyLink" or "termsLink" parameter to the WET
+ * function "wet.build.footer".  Needed because this parameter can be a footer link
+ * or an array of footer links depending on if showFooter is true or false.
+ */
+public class FooterLinkAdapter extends TypeAdapter<FooterLinkContext>{
+    @Override
+    public void write(JsonWriter jw, FooterLinkContext obj) throws IOException {
+
+        if (obj == null) {
+            jw.nullValue();
+            return;
+        }
+
+        //If showFooter is false, render footer link as an array
+        if (!obj.isShowFooter()) jw.beginArray();
+        jw.beginObject();
+        jw.name("href");
+        jw.value(obj.getFooterLink().getHref());
+        jw.name("newWindow");
+        jw.value(obj.getFooterLink().getNewWindow());
+        jw.endObject();
```

Comment:
We should not manually serialize FooterLink here, this can lead to all sort of problems in the future if FooterLink is updated or gets a custom TypeAdapter of its own.  I know at first glance there is no choice because `JsonWriter` doesn't have anything like `write(Object)` method to serialize nested classes but... 

I see two options:
**Option 1:** Create a TypeAdapterFactory for this TypeAdapter which would pass the "gson" object along when creating new FooterLinkAdapter objects.  I'm not very tempted by this option because it is more complex than next option and also requires us to manually register the TypeAdapterFactory.

**Option 2:** Here is what I propose we do:
1. We have to make `AbstractCoreBean.gson` public but if we do that I would prefer not keeping it there.  So I would move it to a new class (but keep it static), let's say we would create "final class" `component.JsonRenderer`. 
2. Then, in `AbstractCoreBean`, to avoid having to refer JsonRenderer everywhere, I would add a static import: `import static goc.webtemplate.component.JsonRenderer.gson`
3. There here, I would replace the code between begin/end array by either `gson.toJson(obj.getFooterLink(), obj.getFooterLink().getClass(), jw)` or `gson.getAdapter(obj.getFooterLink()).write(jw, obj.getFooterLink())`

NOTE: I'm mostly saying this without having tried it, make sure to test thoroughly.
ref: https://stackoverflow.com/questions/42644832/convert-nested-json-to-nested-java-object-using-gson-typeadapter

Reply:
done